pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/BackfillDateTimeColumnCommand.java

```diff
@@ -0,0 +1,232 @@
+  }
+
+
+  public BackfillDateTimeColumnCommand setSegmentNames(String segmentNames) {
+    _segmentNames = segmentNames;
+    return this;
+  }
+
+  public BackfillDateTimeColumnCommand setSegmentType(SegmentType segmentType) {
+    _segmentType = segmentType;
+    return this;
+  }
+
+  public BackfillDateTimeColumnCommand setSrcTimeFieldSpec(String srcTimeFieldSpec) {
+    _srcTimeFieldSpec = srcTimeFieldSpec;
+    return this;
+  }
+
+  public BackfillDateTimeColumnCommand setDestDateTimeFieldSpec(String destDateTimeFieldSpec) {
+    _destDateTimeFieldSpec = destDateTimeFieldSpec;
+    return this;
+  }
+
+  public BackfillDateTimeColumnCommand setBackupDir(String backupDir) {
+    _backupDir = backupDir;
+    return this;
+  }
+
+
+  @Override
+  public String toString() {
+    return ("BackfillSegmentColumn  -controllerHost " + _controllerHost + " -controllerPort " + _controllerPort
+        + " -tableName " + _tableName + " -segmentNames " + _segmentNames + " -segmentType " + _segmentType
+        + " -srcTimeFieldSpec " + _srcTimeFieldSpec + " _destDateTimeFieldSpec " + _destDateTimeFieldSpec
+        + " -backupDir " + _backupDir);
+  }
+
+  @Override
+  public final String getName() {
+    return "BackfillSegmentColumn";
+  }
+
+  @Override
+  public String description() {
+    return "Backfill a column in segments of a pinot table, with a millis value corresponding to the time column";
+  }
+
+  @Override
+  public boolean getHelp() {
+    return _help;
+  }
+
+  @Override
+  public boolean execute() throws Exception {
+    LOGGER.info("Executing command: {}", toString());
+
+
+    if (_controllerHost == null || _controllerPort == null) {
+      throw new RuntimeException("Must specify controller host and port.");
+    }
+
+    if (_srcTimeFieldSpec == null || _destDateTimeFieldSpec == null) {
+      throw new RuntimeException("Must specify srcTimeFieldSpec and destTimeFieldSpec.");
+    }
+    TimeFieldSpec timeFieldSpec = OBJECT_MAPPER.readValue(new File(_srcTimeFieldSpec), TimeFieldSpec.class);
+    DateTimeFieldSpec dateTimeFieldSpec = OBJECT_MAPPER.readValue(new File(_destDateTimeFieldSpec), DateTimeFieldSpec.class);
+
+    if (_tableName == null) {
+      throw new RuntimeException("Must specify tableName.");
+    }
+
+    BackfillSegmentUtils backfillSegmentUtils =
+        new BackfillSegmentUtils(_controllerHost, _controllerPort);
+
+    List<String> segmentNames = new ArrayList<>();
+    List<String> allSegmentNames = backfillSegmentUtils.getAllSegments(_tableName, _segmentType);
+    if (_segmentNames == null) {
+      segmentNames = allSegmentNames;
+    } else {
+      for (String segmentName : _segmentNames.split(",")) {
+        if (allSegmentNames.contains(segmentName)) {
+          segmentNames.add(segmentName);
+        } else {
+          throw new RuntimeException("Segment with name " + segmentName + " does not exist.");
+        }
+      }
+    }
+
+    if (_backupDir == null) {
+      _backupDir = TMP_DIR;
```

Comment:
Not sure where TMP_DIR is coming from, but if backupDir is not defined, maybe you should just not back things up? TMP_DIR may fill up /tmp (if that is the backup dir) and cause a system to become unusable.

Or, maybe set it to /var/tmp which tends to hold some more data

Reply:
Made backup dir mandatory. I prefer having a compulsory backup, and yes better to let user decide where to backup